app/controllers/accounts_controller.rb

```diff
@@ -45,6 +47,13 @@ def archive
 
   private
 
+  def update_kids_info
+    account_params = params.require(:account).permit(:name, :email)
+    current_user.update!(account_params)
```

Comment:
on the line above you check for no current user present. how it can be present here?

Reply:
On the line above I'm checking that the current user is not a parent

update_kids_info unless current_user.**parent?**

Reply:
Oh, sorry ) nevertheless this code looks a bit hard to read. I anticipate issues with its maintenance because of that